gamla/functional_generic.py

```diff
@@ -649,3 +649,17 @@ def groupby(
         ),
         valmap(tuple),
     )
+
+
+def countby(f: Callable) -> Dict[Any, int]:
+    """
+    Count elements of a collection by a function `f`.
+    Return a mapping `{y: len({x s.t. key(x) = y})}.`
+
+    >>> countby(functional.head)(["aa", "ab", "ac", "bc"])
+    {'a': 3, 'b': 1}
+    """
+    return compose_left(
+        groupby(f),
```

Comment:
not very efficinent if the grouped collection is large
consider `groupby_many_reduce`